src/plumpy/base/state_machine.py

```diff
@@ -320,7 +346,8 @@ def transition_to(self, new_state: Union[Hashable, State, Type[State]], *args: A
                 self._enter_next_state(new_state)
             except StateEntryFailed as exception:
                 # Make sure we have a state instance
-                new_state = self._create_state_instance(exception.state, *exception.args, **exception.kwargs)
+                if not isinstance(exception.state, State):
+                    new_state = self._create_state_instance(exception.state, **exception.kwargs)
```

Comment:
The code above is the outdated one? 
The change came with `StateEntryFailed` is now only take the state object and the `exception.state` will sure be an object instead of an state class.